### DISCUSSION TITLE
How to automate editing of Kubernetes YAML files for Github Actions

### DISCUSSION THREAD

**Discussion by siegfried01** [community/community#26319] [Actions]
<p>I would like to deploy this sample Kubernetes app with Github action/workflow…</p>
<p><a href="https://github.com/siegfried01/AzureCosmosDB_AKS" rel="noopener nofollow ugc">https://github.com/siegfried01/AzureCosmosDB_AKS</a></p>
<p>As you can see, we need to (for example) execute an azure CLI command to get azure cosmos MongoDB connection string using azure CLI (or powershell) and paste this in the YAML file… Can someone guide me on how to do this? Maybe there is a similar example somewhere? I was not sure what to bing/google search for.</p>
<p>Thanks</p>
<p>Siegfried</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>For modifying the YAML file the easiest approach is some kind of templating. For simple cases <code>sed</code> is probably enough (it’s a Unix tool, not sure if it’s available in Powershell, but Actions has Bash on Windows, too), put a placeholder into the YAML and do something like</p>
<pre><code class="lang-auto">sed -e s/PLACEHOLDER/YOUR_STRING/g
</code></pre>
<p>to replace every occurrence of <code>PLACEHOLDER</code> with <code>YOUR_STRING</code>.</p>
<p>If you need something more complex look at proper templating engines (like <a href="https://jinja.palletsprojects.com/en/3.1.x/">Jinja</a>), or go all the way to managing your deployment with <a href="https://helm.sh/docs/">Helm</a>.</p>
<p>I’m not familiar with Azure so I can’t tell you how to get that URL, but once you have it you can transfer it between steps as a <a href="https://docs.github.com/en/actions/learn-github-actions/contexts#example-usage-of-the-steps-context">step output</a> or <a href="https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#environment-files">environment variable</a> and use that in your templating command.</p>

**Comment by ViacheslavKudinov** [comment]
<p>Hi,<br>
For AKS can be used <a href="https://github.com/Azure/k8s-deploy" class="inline-onebox" rel="noopener nofollow ugc">GitHub - Azure/k8s-deploy: GitHub Action for deploying to Kubernetes clusters</a><br>
There is also repo with different Azure GHA examples <a href="https://github.com/Azure/actions-workflow-samples" class="inline-onebox" rel="noopener nofollow ugc">GitHub - Azure/actions-workflow-samples: Help developers to easily get started with GitHub Action workflows to deploy to Azure</a></p>
